### PR TITLE
WIP: Initial version of the IPAM plugin

### PR DESCRIPTION
**Pull request by leodotcloud**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,9 +1,64 @@
...
 import (
-	"github.com/Sirupsen/logrus"
+	"fmt"
+	"net"
+	"os"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	//"github.com/rancher/rancher-cni-ipam/fake_allocator"
```
Comment (alena1108) [inline main.go]: remove these comments

File: fake_allocator/backend/disk/lock.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
+// Copyright 2015 CNI authors
```
Comment (alena1108) [inline fake_allocator/backend/disk/lock.go]: should be moved under ip_finder; or remove if we don't use it anywhere

File: main.go [hunk 3]
```diff
@@ -1,9 +1,64 @@
 package main
```
Comment (alena1108) [inline main.go:1]: shall we move it to the top of the project under "tests" dir?

Comment (alena1108) [inline main.go:1]: and move metadata-test binary to the .gitignore

**Comment by leodotcloud** [comment]
Thanks for the comments, I will create a new PR with the changes.

**Pull request closed without merging**